src/components/sidebar/routesView/RouteShow.tsx

```diff
@@ -25,8 +26,13 @@ interface IRouteShowProps {
 @inject('routeStore')
 @observer
 class RouteShow extends React.Component<IRouteShowProps> {
-    constructor(props: IRouteShowProps) {
-        super(props);
+    async componentDidMount() {
+        this.props.route.routePaths.forEach((routePath, index) => {
+            // Make two first route paths visible by default
+            if (index < 2) {
+                this.props.routeStore!.toggleRoutePathVisibility(routePath.internalId);
+            }
+        });
```

Comment:
This is a hack imo. I think its not the component's responsibility to set first two visible. I would set the property visible for the first two routePaths when they are put to the store for the first time if < 2 routePaths are visible.

Nit: also this isn't in the scope of this task.

Edit: hmm, it was factory's job before... but maybe thats not the correct place either...

Reply:
In my opinion this belongs to the component. Before it was, as you noticed in the factory, but the factory should just create the object for us. In my opinion the factory should not need to know even what the visible thing means, it should just deliver a object from Jore.
I think it is the components responsibility to set this variable, in my opinion the component is the only module that actually needs to know what the variable even does. This is because the variable is only used in this specific component, and the visible property is then reflected forward on the map.